File: app/src/main/java/com/googlenews/googlenews/presenter/NewsFetchPresenter.java
package com.googlenews.googlenews.presenter;

import android.content.Context;
import android.util.Log;

import com.googlenews.googlenews.interactore.NewsFetchInteractore;
import com.googlenews.googlenews.model.Article;
import com.googlenews.googlenews.view.NewsInterface;
import com.loopj.android.http.RequestParams;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;
import java.util.ArrayList;

public class NewsFetchPresenter implements  PresenterInterface{
    ArrayList<Article> arrayList;
    NewsInterface mNewsInterface;
    Context mContext;
    NewsFetchInteractore interactore;
    public NewsFetchPresenter(NewsInterface newsInterface, Context context) {
        this.mContext=context;
        this.mNewsInterface=newsInterface;
         interactore=new NewsFetchInteractore(this,mContext);
    }

    @Override
    public void getNews(String url,RequestParams requestParams) {
        interactore.getNewsFormGoogle(url,requestParams);
    }

    @Override
    public void startProgressDialog() {
        mNewsInterface.startProgressDialog();
    }

    @Override
    public void closeProgressDialog() {
        mNewsInterface.closeProgressDialog();
    }

    @Override
    public void newsResponce(byte[] responseBody) throws IOException {
        JSONObject object= null;
        JSONArray jsonArray;
        arrayList=new ArrayList<>();
        try {

            object = new JSONObject(new String(responseBody));
            jsonArray=object.getJSONArray("articles");
            Log.i("ghfg", "newsResponce: "+jsonArray.length());

            for (int i=0; i < jsonArray.length(); i++) {
                JSONObject jpersonObj = jsonArray.getJSONObject(i);
                Article article = new Article();
                article.setUrl(jpersonObj.getString("url"));
                article.setAuthor(jpersonObj.getString("author"));
                article.setDescription(jpersonObj.getString("description"));
                article.setTitle(jpersonObj.getString("title"));
                article.setUrlToImage(jpersonObj.getString("urlToImage"));
                article.setPublishedAt(jpersonObj.getString("publishedAt"));
                arrayList.add(article);
            }

            mNewsInterface.newsResponceFromServer(arrayList);
        } catch (JSONException e) {
            e.printStackTrace();
        }

    }
}

File: app/src/main/java/com/googlenews/googlenews/presenter/NewsFetchPresenter.kt
package com.googlenews.googlenews.presenter

import android.content.Context

import com.googlenews.googlenews.interactore.NewsFetchInteractore
import com.googlenews.googlenews.model.Article
import com.googlenews.googlenews.view.NewsInterface
import org.json.JSONArray

import org.json.JSONObject


class NewsFetchPresenter(var mNewsInterface: NewsInterface, var mContext: Context) : PresenterInterface {
    override fun setResponse(response: JSONObject?) {
        var articaleArray= ArrayList<Article>();
        var jsonArray: JSONArray
        jsonArray = response?.get("articles") as JSONArray
        for (i in 0..jsonArray.length() - 1) {
            var jsonobj: JSONObject
            jsonobj = jsonArray[i] as JSONObject
            var article = Article(
                    author ="",//+ jsonobj.get("author") as String,
                    description = "",//jsonobj.get("description") as String,
                    publishedAt ="",// jsonobj.get("publishedAt") as String,
                    title = jsonobj.get("title") as String,
                    urlToImage = jsonobj.get("urlToImage") as String,
                    url = jsonobj.get("url") as String)
            print(article)
            articaleArray.add(article)
        }


        mNewsInterface.newsResponceFromServer(articaleArray)
    }


    internal var service: NewsFetchInteractore

    init {
        service = NewsFetchInteractore(this, mContext)
    }

    override fun getNews(url: String) {

        service.getNewsFormGoogle(url)

    }

    override fun closeProgressDialog() {
        mNewsInterface.closeProgressDialog()
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun startProgressDialog() {
        mNewsInterface.startProgressDialog()
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

}
